general/cuda.cpp

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2010, Lawrence Livermore National Security, LLC. Produced at
+// the Lawrence Livermore National Laboratory. LLNL-CODE-443211. All Rights
+// reserved. See file COPYRIGHT for details.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability see http://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the GNU Lesser General Public License (as published by the Free
+// Software Foundation) version 2.1 dated February 1999.
+
+#include "okina.hpp"
+
+namespace mfem
+{
+
+void* CuMemAlloc(void** dptr, size_t bytes)
+{
+#ifdef MFEM_USE_CUDA
+   if (CUDA_SUCCESS != ::cuMemAlloc((CUdeviceptr*)dptr, bytes))
+   {
+      mfem_error("Error in CuMemAlloc");
```

Comment:
I think we should return the Cuda error instead of a simple `Error in CuMemAlloc`. Same for the following Cuda errors.